tests/harness/cc/BUILD

```diff
@@ -1,19 +1,50 @@
+C_OPTS = [
+    "-Wall",
+    "-Wextra",
+    "-Werror",
+    "-Wnon-virtual-dtor",
+    "-Woverloaded-virtual",
+    "-Wold-style-cast",
+    "-std=c++14",
+]
+
 cc_binary(
     name = "cc-harness",
     srcs = ["harness.cc"],
-    visibility = ["//visibility:public"],
     # These ensure that we are at least compatible with what Envoy is expecting.
-    copts = [
-        "-Wall",
-        "-Wextra",
-        "-Werror",
-        "-Wnon-virtual-dtor",
-        "-Woverloaded-virtual",
-        "-Wold-style-cast",
-        "-std=c++14",
-    ],
+    copts = C_OPTS,
+    visibility = ["//visibility:public"],
     deps = [
         "//tests/harness:harness_proto",
         "//tests/harness/cases:cc",
     ],
 )
+
+# Ensure that if the headers are included in multiple libraries, those libraries
+# can be linked without conflicts.
+cc_test(
+    name = "cc-diamond-test",
+    srcs = ["diamond_test.cc"],
+    copts = C_OPTS,
+    linkstatic = 1,  # Forces both libraries to be linked in. DO NOT REMOVE THIS
+    deps = [
+        "cc-diamond-0",
+        "cc-diamond-1",
+    ],
+)
+
+cc_library(
+    name = "cc-diamond-0",
```

Comment:
Nit: prefer `_` over `-` when naming targets. This is mostly for consistency, but can also be a real issue when working with some languages, e.g. Python doesn't like modules with `-`.